pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -56,6 +56,18 @@ func (l *Prometheus) GetStatus() PrometheusStatus {
 	return l.Status
 }
 
+// +kubebuilder:validation:Enum=name;shard
```

Comment:
only a suggestion but I feel that name and shard are a bit vague.
```suggestion
// +kubebuilder:validation:Enum=ResourceName;ShardName
```